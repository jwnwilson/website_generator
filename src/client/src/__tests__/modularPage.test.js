/* eslint-env jest */

import { shallow, render } from 'enzyme'
import React from 'react'
import { useStaticQuery } from "gatsby"
import renderer from 'react-test-renderer'

import PageList from '../templates/modularPage';

const testData = {page: {content: JSON.stringify({modules: []})}};

const setup = () => {
}

beforeEach(() => {
  // Use this for more complicated tests and logic
  // useStaticQuery.mockImplementationOnce(({ render }) =>
  //   render({
  //     site: {
  //       siteMetadata: {
  //         title: `Default Starter`,
  //       },
  //     },
  //   })
  // )
  useStaticQuery.mockReturnValue(
    {
      site: {
        siteMetadata: {
          title: `Test Title`,
        },
      },
    }
  );
})

describe('With Enzyme', () => {
  it('App shows "Modular static website"', () => {
    setup();

    const pageList = render(<PageList data={testData}/>)
    expect(pageList.find('h1').text()).toEqual('Test Title')
  })
})

describe('With Snapshot Testing', () => {
  it('App shows "Modular static website"', () => {
    setup();
    const component = renderer.create(<PageList data={testData}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})