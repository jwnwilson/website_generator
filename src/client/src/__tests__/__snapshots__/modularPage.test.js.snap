// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Modular static website" 1`] = `
Array [
  <header
    style={
      Object {
        "background": "rebeccapurple",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          Test Title
        </a>
      </h1>
    </div>
  </header>,
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0px 1.0875rem 1.45rem",
        "paddingTop": 0,
      }
    }
  >
    <main />
    <footer>
      Â© 
      2019
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>,
]
`;
